mariadb connector error

### Real behavior (Required)
```
java.sql.SQLNonTransientConnectionException: Could not connect to address=(host=localhost)(port=9030)(type=master) : (conn=0) could not load system variables

	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:73)
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:197)
	at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.connectWithoutProxy(AbstractConnectProtocol.java:1394)
	at org.mariadb.jdbc.internal.util.Utils.retrieveProxy(Utils.java:635)
	at org.mariadb.jdbc.MariaDbConnection.newConnection(MariaDbConnection.java:150)
	at org.mariadb.jdbc.Driver.connect(Driver.java:89)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at com.starrocks.pseudocluster.PseudoClusterTest.testCreateTabletAndInsert(PseudoClusterTest.java:34)
Caused by: java.sql.SQLNonTransientConnectionException: (conn=0) could not load system variables
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:73)
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:188)
	at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.requestSessionDataWithShow(AbstractConnectProtocol.java:1114)
	at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.additionalData(AbstractConnectProtocol.java:1127)
	at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.postConnectionQueries(AbstractConnectProtocol.java:895)
	at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.createConnection(AbstractConnectProtocol.java:600)
	at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.connectWithoutProxy(AbstractConnectProtocol.java:1389)
	... 6 more
Caused by: java.sql.SQLNonTransientConnectionException: (conn=0) could not load system variables. socket connected: true
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:73)
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:184)
	at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.requestSessionDataWithShow(AbstractConnectProtocol.java:1107)
	... 10 more

possibly we should support these SQL statements:

  private static final byte[] SESSION_QUERY =
      ("SELECT @@max_allowed_packet,"
              + "@@system_time_zone,"
              + "@@time_zone,"
              + "@@auto_increment_increment")
          .getBytes(StandardCharsets.UTF_8);
  private static final byte[] IS_MASTER_QUERY =
      "select @@innodb_read_only".getBytes(StandardCharsets.UTF_8);

          "SHOW VARIABLES WHERE Variable_name in ("
              + "'max_allowed_packet',"
              + "'system_time_zone',"
              + "'time_zone',"
              + "'auto_increment_increment')");
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

--- a/fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java
+++ b/fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java
@@ -110,6 +110,7 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
     public static final String NET_WRITE_TIMEOUT = "net_write_timeout";
     public static final String NET_READ_TIMEOUT = "net_read_timeout";
     public static final String TIME_ZONE = "time_zone";
+    public static final String INNODB_READ_ONLY = "innodb_read_only";
     public static final String SQL_SAFE_UPDATES = "sql_safe_updates";
     public static final String NET_BUFFER_LENGTH = "net_buffer_length";
     public static final String CODEGEN_LEVEL = "codegen_level";
@@ -425,6 +426,9 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
     // The current time zone
     @VariableMgr.VarAttr(name = TIME_ZONE)
     private String timeZone = TimeUtils.getSystemTimeZone().getID();
+
+    @VariableMgr.VarAttr(name = INNODB_READ_ONLY)
+    private boolean innodbReadOnly = true;
 
     @VariableMgr.VarAttr(name = PARALLEL_EXCHANGE_INSTANCE_NUM)
     private int exchangeInstanceParallel = -1;
@@ -745,6 +749,14 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
 
     public void setTimeZone(String timeZone) {
         this.timeZone = timeZone;
+    }
+
+    public boolean isInnodbReadOnly() {
+        return innodbReadOnly;
+    }
+
+    public void setInnodbReadOnly(boolean innodbReadOnly) {
+        this.innodbReadOnly = innodbReadOnly;
     }
 
     public void setMaxExecMemByte(long maxExecMemByte) {

--- a/fe/fe-core/src/test/java/com/starrocks/qe/VariableMgrTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/qe/VariableMgrTest.java
@@ -90,6 +90,7 @@ public class VariableMgrTest {
         Assert.assertEquals(300, var.getQueryTimeoutS());
         Assert.assertEquals(false, var.isReportSucc());
         Assert.assertEquals(32L, var.getSqlMode());
+        Assert.assertEquals(true, var.isInnodbReadOnly());
 
         List<List<String>> rows = VariableMgr.dump(SetType.SESSION, var, null);
         Assert.assertTrue(rows.size() > 5);

